docs/containers-auth.json.5.md

```diff
@@ -7,7 +7,10 @@ containers-auth.json - syntax for the registry authentication file
 
 A credentials file in JSON format used to authenticate against container image registries.
 On Linux it is stored at `${XDG_RUNTIME_DIR}/containers/auth.json`;
-on Windows and macOS, at `$HOME/.config/containers/auth.json`
+on Windows and macOS, at `$HOME/.config/containers/auth.json`.
+
+When searching for the credential for a registry, the following files will be read in sequence until the valid credential is found: the system-wide file will be first searched if configured, if credentials are not present, search in `${XDG\_RUNTIME\_DIR}/containers/auth.json`, `${XDG\_CONFIG\_HOME}/containers/auth.json`, `$HOME/.docker/config.json`.
```

Comment:
- It took me multiple tries to realize “the system-wide file … if configured” refers the `SystemContext.AuthFilePath` and the like; this should probably refer to an explicit override using an option of the calling application, or something like that (end users don’t interact with `SystemContext` directly).
- If the above _is_ configured, `XDG_RUNTIME_DIR/…` is not attempted.
- This ultimately should clearly differentiate between the one file that the code reads _and_ writes, and the others than are read-only (unless the user explicitly points at it as an override).

---

Using `os.UserConfigDir` is very interesting, but the results are not the same as `XDG_RUNTIME_DIR` on non-Linux platforms. At this point I’m not sure we have the expertise or the time/interest to commit to adding another Windows/macOS … directory (and we can do so later if the expertise/interest comes up); so maybe only add the new behavior “on traditional Unix” (i.e. if we are not using the `nonLinuxAuthFilePath` value).